### PR TITLE
Adding OAI RAN and UE test deployment scripts

### PR DESCRIPTION
**Pull request by josephthaliath**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> 
kind feature
> /kind flake

**What this PR does / why we need it**:

This PR is adding new test scripts for deploying OAI RAN and core NFs

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
New test scripts are included for deploying OAI RAN and core NFs using Nephio

```

### CODE REVIEW

**Comment by johnbelamaric** [comment]
/ok-to-test

I also fixed it so that won't be needed anymore (gave the sig-automation team Triage access on this repo)

File: e2e/tests/oai/002.sh [hunk 1]
```diff
@@ -27,18 +27,26 @@ source "${LIBDIR}/k8s.sh"
 # shellcheck source=e2e/lib/kpt.sh
 source "${LIBDIR}/kpt.sh"
 
+#adding directory path to register repository to have upstream packages in repository subdirectory when used with packagevariantset
+#TODO: to change the private catalog repository to nephio catalog repository after PR approval
+kpt alpha repo register --namespace default https://github.com/josephthaliath/catalog.git --directory=workloads/oai --name=oai-ran-packages
```
Comment (electrocucaracha) [inline e2e/tests/oai/002.sh:32]: @josephthaliath are you using this [PR](https://github.com/nephio-project/catalog/pull/11) for these tests?

File: e2e/tests/oai/002-oai-operators.yaml [hunk 2]
```diff
@@ -64,3 +64,20 @@ spec:
...
+    repo: oai-ran-packages
+    package: oai-ran-operator
+    revision: v1
+  targets:
+  - objectSelector:
+      apiVersion: infra.nephio.org/v1alpha1
+      kind: WorkloadCluster
+    template:
+      annotations:
+        approval.nephio.org/policy: initial
```
Comment (electrocucaracha) [inline e2e/tests/oai/002-oai-operators.yaml:83]: It seems like you have a line ending at in the file, maybe it's added by your text editor

File: e2e/tests/oai/002-oai-operators.yaml [hunk 3]
```diff
@@ -64,3 +64,20 @@ spec:
...
+---
+apiVersion: config.porch.kpt.dev/v1alpha2
+kind: PackageVariantSet
+metadata:
+  name: oai-ran-operator
+spec:
+  upstream:
+    repo: oai-ran-packages
+    package: oai-ran-operator
+    revision: v1
```
Comment (electrocucaracha) [inline e2e/tests/oai/002-oai-operators.yaml:76]: Are you sure that this is the right revision?
```suggestion
    revision: r2
```

File: e2e/tests/oai/002.sh [hunk 4]
```diff
@@ -27,18 +27,26 @@ source "${LIBDIR}/k8s.sh"
...
+kpt_wait_pkg "edge" "oai-ran-operator"
 
 _core_kubeconfig="$(k8s_get_capi_kubeconfig "core")"
 k8s_wait_ready_replicas "deployment" "mysql" "$_core_kubeconfig" "oai-core"
 for controller in amf ausf nrf smf udm udr; do
     k8s_wait_ready_replicas "deployment" "oai-$controller-controller" "$_core_kubeconfig" "oai-operators"
 done
 k8s_wait_ready_replicas "deployment" "oai-upf-controller" "$(k8s_get_capi_kubeconfig "edge")" "oai-operators"
+k8s_wait_ready_replicas "deployment" "oai-ran-operator" "$(k8s_get_capi_kubeconfig "regional")" "oai-ran-operators"
+k8s_wait_ready_replicas "deployment" "oai-ran-operator" "$(k8s_get_capi_kubeconfig "edge")" "oai-ran-operators"
```
Comment (electrocucaracha) [inline e2e/tests/oai/002.sh:52]: Maybe we can store the value of kubeconfig of the edge and reuse it.
```suggestion
_edge_kubeconfig="$(k8s_get_capi_kubeconfig "edge")"
k8s_wait_ready_replicas "deployment" "oai-upf-controller" "$_edge_kubeconfig" "oai-operators"
k8s_wait_ready_replicas "deployment" "oai-ran-operator" "$(k8s_get_capi_kubeconfig "regional")" "oai-ran-operators"
k8s_wait_ready_replicas "deployment" "oai-ran-operator" "$_edge_kubeconfig" "oai-ran-operators"
```

File: e2e/tests/oai/004.sh [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    package: pkg-example-cucp-bp
+    revision: v1
+  downstream:
+    repo: regional
+    package: oai-ran-cucp
+  annotations:
+    approval.nephio.org/policy: initial
+  injectors:
+  - name: regional
+EOF
```
Comment (electrocucaracha) [inline e2e/tests/oai/004.sh:101]: I think that we can create a separate file for this.

File: e2e/tests/oai/004.sh [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+source "${LIBDIR}/k8s.sh"
+
+# shellcheck source=e2e/lib/kpt.sh
+source "${LIBDIR}/kpt.sh"
+
+# shellcheck source=e2e/lib/porch.sh
+source "${LIBDIR}/porch.sh"
+
+# shellcheck source=e2e/lib/_assertions.sh
+source "${LIBDIR}/_assertions.sh"
```
Comment (electrocucaracha) [inline e2e/tests/oai/004.sh:34]: I don't think that the functions stored in these files are being used

File: e2e/tests/oai/005.sh [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+source "${LIBDIR}/k8s.sh"
+
+# shellcheck source=e2e/lib/kpt.sh
+source "${LIBDIR}/kpt.sh"
+
+# shellcheck source=e2e/lib/porch.sh
+source "${LIBDIR}/porch.sh"
+
+# shellcheck source=e2e/lib/_assertions.sh
+source "${LIBDIR}/_assertions.sh"
```
Comment (electrocucaracha) [inline e2e/tests/oai/005.sh:34]: Same thing here, I don't think those are being used

File: e2e/tests/oai/005.sh [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    package: pkg-example-ue-bp
+    revision: v1
+  downstream:
+    repo: edge
+    package: oai-ran-ue
+  annotations:
+    approval.nephio.org/policy: initial
+  injectors:
+  - name: edge
+EOF
```
Comment (electrocucaracha) [inline e2e/tests/oai/005.sh:85]: We can keep this code in a separate yaml file

**Review by electrocucaracha — CHANGES_REQUESTED** [review]
@josephthaliath I made few suggestions on these files, don't forget to add information into the description of the PR

**Comment by electrocucaracha** [comment]
@josephthaliath could you fill some information in the PR's description? I'll be reviewing the errors

**Comment by josephthaliath** [comment]
> @josephthaliath could you fill some information in the PR's description? I'll be reviewing the errors

@electrocucaracha , I have updated PR description

File: e2e/tests/oai/002-oai-operators.yaml [hunk 9]
```diff
@@ -64,3 +64,22 @@ spec:
...
   - name: edge
+---
+apiVersion: config.porch.kpt.dev/v1alpha2
+kind: PackageVariantSet
+metadata:
+  name: oai-ran-operator
+spec:
+  upstream:
+    repo: catalog
+    package: workloads/oai/oai-ran-operator
```
Comment (electrocucaracha) [inline e2e/tests/oai/002-oai-operators.yaml:75]:
```suggestion
    repo: catalog-workloads-oai-ran
    package: oai-ran-operator
```

File: e2e/tests/oai/004-ran-cucp.yaml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+##############################################################################
+
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: oai-cucp
+spec:
+  upstream:
+    repo: catalog
+    package: workloads/oai/pkg-example-cucp-bp
```
Comment (electrocucaracha) [inline e2e/tests/oai/004-ran-cucp.yaml:18]:
```suggestion
    repo: catalog-workloads-oai-ran
    package: pkg-example-cucp-bp
```

File: e2e/tests/oai/004-ran-cucp.yaml [hunk 11]
```diff
@@ -0,0 +1,26 @@
+---
+# SPDX-license-identifier: Apache-2.0
+##############################################################################
+# Copyright (c) 2023 The Nephio Authors.
```
Comment (electrocucaracha) [inline e2e/tests/oai/004-ran-cucp.yaml:4]:
```suggestion
# Copyright (c) 2024 The Nephio Authors.
```

File: e2e/tests/oai/004.sh [hunk 12]
```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+#!/usr/bin/env bash
+# SPDX-license-identifier: Apache-2.0
+##############################################################################
+# Copyright (c) 2023 The Nephio Authors.
```
Comment (electrocucaracha) [inline e2e/tests/oai/004.sh:5]:
```suggestion
# Copyright (c) 2024 The Nephio Authors.
```

File: e2e/tests/oai/004.sh [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+for nf in du cuup; do
+    cat <<EOF | kubectl apply -f -
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: oai-$nf
+spec:
+  upstream:
+    repo: catalog
+    package: workloads/oai/pkg-example-$nf-bp
```
Comment (electrocucaracha) [inline e2e/tests/oai/004.sh:79]:
```suggestion
    repo: catalog-workloads-oai-ran
    package: pkg-example-$nf-bp
```

File: e2e/tests/oai/005.sh [hunk 14]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+#!/usr/bin/env bash
+# SPDX-license-identifier: Apache-2.0
+##############################################################################
+# Copyright (c) 2023 The Nephio Authors.
```
Comment (electrocucaracha) [inline e2e/tests/oai/005.sh:5]:
```suggestion
# Copyright (c) 2024 The Nephio Authors.
```

File: e2e/tests/oai/005.sh [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+cat <<EOF | kubectl apply -f -
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: oai-ue
+spec:
+  upstream:
+    repo: catalog
+    package: workloads/oai/pkg-example-ue-bp
```
Comment (electrocucaracha) [inline e2e/tests/oai/005.sh:69]:
```suggestion
    repo: catalog-workloads-oai-ran
    package: pkg-example-ue-bp
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]
These suggestions require merging the https://github.com/nephio-project/test-infra/pull/230

**Comment by electrocucaracha** [comment]
@josephthaliath did you rebase this PR with the latest changes?

**Comment by electrocucaracha** [comment]
It seems like the tests are passing the 75 min threshold, so we will need [this change](https://github.com/nephio-project/test-infra/pull/230/commits/b27db45fb50e4a28d0d86f7b9e2998c517304b22) to increase the timeout

**Comment by arora-sagar** [comment]
@josephthaliath Can you avoid deploying RAN operator on the core cluster if possible?

**Comment by arora-sagar** [comment]
@josephthaliath @electrocucaracha I merged this with main branch and everything went well. The end to end test was working fine. Here you can see some numbers. 

```
./init.sh --> 12 mins
001.sh --> 20:49:23 -- 21:06:04 --> 15 mins
002.sh --> 21:06:58 -- 21:09:59 --> 3 mins
003.sh --> 21:11:37 -- 21:42:26 --> 31 mins 
004.sh --> 22:00:46 -- 22:24:15 --> 24 mins
005.sh --> 22:26:10 -- 22:27:43 --> 1 min
```

I think the timeout for end to end test should be 100 or 120 mins to be on the safe side. We can reduce it if we see it takes less time.

File: test-infra_18jan_v1.diff
Comment (arora-sagar) [inline test-infra_18jan_v1.diff:1]: @josephthaliath I don't think this file is needed

Comment (josephthaliath) [reply test-infra_18jan_v1.diff:1]: @arora-sagar . Thanks. added by mistake. removed it.

**Pull request merged**